### PR TITLE
#992: Add support for Count Min Sketch to DiceDB

### LINKED ISSUE #992: Add support for Count Min Sketch
DiceDB currently doesn't support the Count Min Sketch data structure. I would like to add `CMS.INITBYDIM`, `CMS.INITBYPROB`, `CMS.INCRBY`, `CMS.QUERY`, `CMS.MERGE`, `CMS.INFO` commands.

I have started looking at the theory and some available implementations.

### PR DESCRIPTION
**Pull request by hgupta12**
This PR adds support for Count Min Sketch (CMS) commands to DiceDB.

- [x] Implement all the commands mentioned in the issue
- [x] Add detailed comments
- [x] Format error messages properly
- [x] Write unit tests for all the commands
- [x] Write integration tests for the commands
- [x] Write http integration tests for the commands

Resolves #992

### CODE REVIEW

**Comment by JyotinderSingh** [comment]
Please add integration and unit tests for these changes, will review once done.

**Comment by hgupta12** [comment]
Hey @JyotinderSingh, I have added the unit and integration tests. Please review it.

**Review by JyotinderSingh — CHANGES_REQUESTED** [review]
Thanks for this massive effort in implementing CMS @hgupta12! The changes largely look good. Have left some small comments around testing.

There is one thing I am concerned about. Async server is soon going to be deprecated and we will need to migrate these new commands to the multi threaded resp server. It would be great if we introduced these methods directly as a part of the store_eval.go file and marked these commands as singleShard commands. What do you think? Imo this will save us a lot of effort in the future in migrating this code.

File: internal/eval/countminsketch_test.go [hunk 1]
```diff
@@ -0,0 +1,318 @@
...
+
+				source1 := "test"
+				opts, _ = newCountMinSketchOpts([]string{"1000", "5"})
+				createCountMinSketch(source1, opts, store)
+
+			},
+			input: []string{"cms_key5", "1", "test"},
+			output: clientio.RespOK,
+		},
+		"correct case with given weights": {
```
Comment (JyotinderSingh) [inline internal/eval/countminsketch_test.go:298]: for ease of debugging in CI logs, these test case names should include the word CMS.

**Comment by hgupta12** [comment]
Hey @JyotinderSingh thanks a lot for the review! I have made the requested changes. As for the commands, I agree with you. I'll make the necessary changes to introduce these commands directly in the resp server.  Just a few doubts -

**Comment by JyotinderSingh** [comment]
Some tests are failing.

**Comment by hgupta12** [comment]
Hey @JyotinderSingh. Yes the http integration tests are failing because the output is not getting parsed properly. I haven't been able to figure out the issue yet but I am looking at other migrations to figure out what am I missing. Meanwhile, if you could take a lot at the issue too that would really helpful. Thank you.

File: internal/eval/countminsketch.go [hunk 2]
```diff
@@ -0,0 +1,538 @@
...
+		}
+		sources = append(sources, c)
+	}
+
+	if len(args) == int(2+numberOfKeys) {
+		weights := slices.Repeat([]uint64{1}, int(numberOfKeys))
+		destination.mergeMatrices(sources, weights, args[0], keys)
+
+		return &EvalResponse{
+			Result: clientio.RespOK,
```
Comment (c-harish) [inline internal/eval/countminsketch.go:301]: Hi @hgupta12. Since your commands are migrated, return clientio.OK instead of clientio.RespOK.

Comment (hgupta12) [reply internal/eval/countminsketch.go]: Yes this worked! Thanks a lot for the review.

**Comment by JyotinderSingh** [comment]
Banger PR, going through it.

**Review by JyotinderSingh — APPROVED** [review]
Thanks for adding support for the CMS data structure @hgupta12! The code looks well structured and well tested. LGTM.

**Pull request merged**